loli/medpy
# Release 0.5.0

# Pre:
- [X] Write a pre-release checklist for future releases and add to repo as documentation

# TODO
- [X] Bump-up library version
- [X] Supported formats
    - `python3 tests/support.py > myformats.log` fails due to random package in scipy
- [X] Depreciation and error fixes
    - scipy. => numpy.*
    - list(slice) => tuple(slice)
- [X] Run tests and fix errors:
    - switch from nose to pytest
    - add test requirements.txt: pytest-7.4.3 & hypothesis-6.92.0
    - fix errors
- [ ] Run all notebooks, to test the code
    - fix all error that occurr
- [ ] Check Readme.md
    - check consistency and update
- [ ] Check documentation
    - https://loli.github.io/medpy/
    - Add graphcut help for https://github.com/loli/medpy/issues/103 CentOs
    - Add note that, under Debian / Ubuntu, lib can be found under /usr/lib/x86_64-linux-gnu/libboost_python* if linker error
    - Generally, write better error handling for GraphCut installation and link in open issues on the topic for people to find (keep open)
    - Issue of broken links and notebooks: https://github.com/loli/medpy/issues/97
- [ ] Check badges in README.md
- [ ] Bioconda / conda
    - Now in conda-forge: https://anaconda.org/conda-forge/medpy
    - test install and update docu (e.g.  https://loli.github.io/medpy/installation/conda.html) accordingly
    - edit recipes to ppoint to 0.5.0?
        - maybe this: https://github.com/conda-forge/medpy-feedstock ?
        - seem like it, version is here: https://github.com/conda-forge/medpy-feedstock/blob/main/recipe/meta.yaml
- [ ] Remove travis build
    - move to github builds, if possible
- [ ] Check config files:
    - .bettercodehub
    - .gitmodules
- [ ] Try installation on different systems
    - with different library versions to figure out minimal
    - with graph cut support and without
    - use `pip install -v -e .` to see installation logs for graphcut
- [ ] check why they exists and remove/ edit requirements-dev.* files in root folder
- [ ] Dockerfiles: won't do anymore
    - remove linked repo
    - remove from documentation
    - Maybe write a short docu / docstring comment on this: https://github.com/loli/medpy/issues/76
    
# new minimal dependency versions
- old dtypes depreciates from numpy >= 1.24
- inspect.getfullargspec recommended since python >= 3.4
- [ ] dump minima dependencies in setup.py

# Optional
- [ ] setup.py: Move from install_requires to requirements.txt?

# Release TODO
- [ ] Generate docs and place to upload
- [ ] Update setup.py
- [ ] Update CHANGES.txt
- [ ] Update github-pages
- [ ] create package and upload to pypi

# Post-Release to do
- [ ] Update conda-forge recipe
